server/dumpbagserver/templates/dumps.html

```diff
@@ -2,70 +2,118 @@
-      </span>
-    </li>
-    {% endfor %}
-  </ul>
-  {% endfor %}
+{% if result is defined %}
+  {% for database, dumps in result | dictsort %}
+     <h2 class="dump-title" onclick="toggleGroup('dump-list-for-{{database}}', '{{database}}')">
+           <a name="{{database}}"></a>{{database}}
+         </h2>     <ul class="mdl-list dump-list" id="dump-list-for-{{database}}"
+           style="display: none">
 
-  <script>
+         {% for dump in dumps | sort(reverse=True) %}
+         <li class="mdl-list__item-two-line">
+           <span
+             onclick="toggleVisibility('dumpinfo-box-{{dump}}')"
+             class="mdl-list__item-primary-content mdl-button mdl-js-button">
+             {{dump}}
+           </span>
+           <span class="mdl-list__item-secondary-content">
+             {{ dump | date_from_dumpname }} UTC
+           </span>
+           <span id="dumpinfo-box-{{dump}}" class="mdl-list__item-secondary-content dumpinfo-box" style="display: none">
+             <strong>Commands</strong>
+             <pre>{{download_commands(database, dump)}}</pre>
+             <a
+               title="Download {{dump}}"
+               class="mdl-button mdl-js-button mdl-button--colored mdl-button--raised mdl-js-ripple-effect"
+               href="{{ url_for('download_dump', db=database, filename=dump) }}"
+               >
+               Download
+             </a>
+           </span>
+         </li>
+         {% endfor %}
+       </ul>
+   {% endfor %}
+   {% else %}
+        {% for database, dumps in dumps | dictsort %}
+          <hr class="dump-list-divider">
+             <h2 class="dump-title" onclick="toggleGroup('dump-list-for-{{database}}', '{{database}}')">
+               <a name="{{database}}"></a>{{database}}
+             </h2>  <ul class="mdl-list dump-list" id="dump-list-for-{{database}}"
+               style="display: none">
+             {% for dump in dumps | sort(reverse=True) %}
+             <li class="mdl-list__item-two-line">
+               <span
+                 onclick="toggleVisibility('dumpinfo-box-{{dump}}')"
+                 class="mdl-list__item-primary-content mdl-button mdl-js-button">
+                 {{dump}}
+               </span>
+               <span class="mdl-list__item-secondary-content">
+                 {{ dump | date_from_dumpname }} UTC
+               </span>
+               <span id="dumpinfo-box-{{dump}}" class="mdl-list__item-secondary-content dumpinfo-box" style="display: none">
+                 <strong>Commands</strong>
+                 <pre>{{download_commands(database, dump)}}</pre>
+                 <a
+                   title="Download {{dump}}"
+                   class="mdl-button mdl-js-button mdl-button--colored mdl-button--raised mdl-js-ripple-effect"
+                   href="{{ url_for('download_dump', db=database, filename=dump) }}"
+                   >
+                   Download
+                 </a>
+               </span>
+             </li>
```

Comment:
This whole block is duplicated. Maybe use another template included in the 2 places for this block?